content/blog/2018-08-28-birds-ocr.md

```diff
@@ -0,0 +1,411 @@
+output](https://github.com/tesseract-ocr/tesseract/wiki/ImproveQuality).
+The [`hocr` package](https://github.com/dmi3kno/hocr) might be of
+interest for post-processing of OCR results.
+
+Below, we only changed the [page segmentation
+mode](https://github.com/tesseract-ocr/tesseract/wiki/ImproveQuality#page-segmentation-method)
+choosing 1 for “Automatic page segmentation with Orientation and script
+detection (OSD)”. We tried using Latin instead of English training data,
+but the results were less good.
+
+One can use either `tesseract::ocr` for a file path, url, or raw vector
+to image, or `magick::image_ocr` for a `magick` object which is quite
+handy in our pipeline.
+
+The function below also filter results of the OCR using language
+detection. By only keeping text recognized as either Latin or English by
+one of the two languages detection packages `cld2` and `cld3` that are
+interfaces to Google compact language detectors 2 and 3, we have a first
+quality filter. If we didn’t do that, we’d end up with more unusable
+text.
+
+``` r
+get_names <- function(image){
+  filename <- attr(image, "filename")
+  ocr_options <- list(tessedit_pageseg_mode = 1)
+   
+  text <- magick::image_ocr(image, options = ocr_options)
+  text <- stringr::str_split(text, "\n", simplify = TRUE)
+  text <- stringr::str_remove_all(text, "[0-9]")
+  text <- stringr::str_remove_all(text, "[:punct:]")
+  text <- trimws(text)
+  text <- stringr::str_remove_all(text, "~")
+  text <- text[text != ""]
+  text <- tolower(text)
+  
+  # remove one letter words
+  # https://stackoverflow.com/questions/31203843/r-find-and-remove-all-one-to-two-letter-words
+  text <- stringr::str_remove_all(text, " *\\b[[:alpha:]]{1,2}\\b *")
+  text <- text[text != ""]
+  
+  # keep only the words that are recognized as either Latin
+  # or English by cld2 or cld3
+  if(length(text) > 0){
+    results <- tibble::tibble(text = text,
+                 cld2 = cld2::detect_language(text),
+                 cld3 = cld3::detect_language(text),
+                 filename = filename)
+  
+  results[results $cld2 %in% c("la", "en") |
+          results$cld3 %in% c("la", "en"),]
+  }else{
+    return(NULL)
+  }
+  
+  
+}
+
+(results1 <- filenames[1] %>%
+  magick::image_read() %>%
+  get_names())
+```
+
+    ## NULL
+
+``` r
+(results2 <- filenames[1] %>%
+  crop_bird() %>%
+  get_names())
+```
+
+    ## # A tibble: 2 x 4
+    ##   text                 cld2  cld3  filename                          
+    ##   <chr>                <chr> <chr> <fs::path>                        
+    ## 1 climacteris picumnus <NA>  la    birds/n115_w1150_42399797481_o.jpg
+    ## 2 brown tree creeper   en    <NA>  birds/n115_w1150_42399797481_o.jpg
+
+So we see that the image transformation was quite useful.
+
+Now, these names look fine, but how to be sure they’re actually
+taxonomic names?
+
+#### Taxonomic name resolution
+
+The [`taxize` package](https://github.com/ropensci/taxize/) by Scott
```

Comment:
with Zachary Foster now having maintainer status too, should he be mentioned even though Scott is major committer?

Reply:
👍

Reply:
Ah no I re-checked taxize's maintainer is @sckott, it's `taxa` that got a new maintainer.

Reply:
np, it was a good remark